h5p-bildetema/src/components/Bildetema/Bildetema.tsx

```diff
@@ -1,15 +1,91 @@
 import React from "react";
-import styles from "./Bildetema.module.scss";
+import { Route, Routes, useNavigate } from "react-router-dom";
 import { Header } from "..";
+import { Language, Topic } from "../../../../common/types/types";
+import { fetchData, getTopics } from "../../../../common/utils/data.utils";
+import { TopicGrid } from "../TopicGrid/TopicGrid";
+import styles from "./Bildetema.module.scss";
+
+type BildetemaProps = {
+  currentLanguage: Language;
+};
+
+export const Bildetema: React.FC<BildetemaProps> = ({ currentLanguage }) => {
+  const [topics, setTopics] = React.useState<Topic[]>([]);
+  const navigate = useNavigate();
+
+  React.useEffect(() => {
+    const run = async (): Promise<void> => {
+      await fetchData();
+      setTopics(await getTopics());
+      navigate(`/${currentLanguage.code}`);
+    };
+    run();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentLanguage.code]);
```

Comment:
Maybe we should try using [react-query](https://react-query.tanstack.com/overview) for async tasks?